Public/Sdk/Experimental/NodeJs/npm.dsc

```diff
@@ -5,33 +5,35 @@ import {Artifact, Cmd, Tool, Transformer} from "Sdk.Transformers";
 
 namespace Npm {
 
-    @@public
-    export function install(args: Arguments) : Result {
-        const folder = Context.getNewOutputDirectory(`npm-${args.name}`);
-        const nodeModulesPath = d`${folder}/node_modules`;
-        const npmCachePath = d`${folder}/npm-cache`;
+    function _install(workingDirectory: StaticDirectory, command: string, ...cmdArgs: Argument[]) : Result {
+        const wd = workingDirectory !== undefined
+            ? workingDirectory.root
+            : Context.getNewOutputDirectory(`npm-${command}`);
+        const nodeModulesPath = d`${wd}/node_modules`;
+        const npmCachePath = Context.getNewOutputDirectory('npm-install-cache');
 
         const arguments: Argument[] = [
             Cmd.argument(Artifact.input(Node.npmCli)),
             Cmd.argument("install"),
-            Cmd.argument(`${args.name}@${args.version}`),
+            ...cmdArgs,
             Cmd.argument("--no-save"), // Prevents writing json files
             Cmd.argument("--no-package-lock"), // Prevents writing json files

```

Comment:
make `--no-package-lock` configurable so that we can generate the package lock when compiling our vscode extension